File: DalatS/app/src/main/java/com/example/dalats/model/CreateIncidentDTO.java
package com.example.dalats.model;

public class CreateIncidentDTO {
    private String title;
    private String description;
    private String address;
    private String ward;
    private String streetName;
    private double latitude;    // Tọa độ sự cố (chọn trên map/nơi ghim)
    private double longitude;
    private int categoryId;

    // --- THÊM MỚI ---
    private double deviceLatitude;  // Tọa độ GPS thực tế của điện thoại
    private double deviceLongitude;
    private boolean isForceCreate;  // Cờ xác nhận (true = gửi bất chấp)

    // Cập nhật Constructor
    public CreateIncidentDTO(String title, String description, String address, String ward, String streetName,
                             double latitude, double longitude, int categoryId,
                             double deviceLatitude, double deviceLongitude, boolean isForceCreate) {
        this.title = title;
        this.description = description;
        this.address = address;
        this.ward = ward;
        this.streetName = streetName;
        this.latitude = latitude;
        this.longitude = longitude;
        this.categoryId = categoryId;

        // Gán giá trị mới
        this.deviceLatitude = deviceLatitude;
        this.deviceLongitude = deviceLongitude;
        this.isForceCreate = isForceCreate;
    }
}

File: SafeDalat_API/SafeDalat_API/Model/DTO/Incident/CreateIncidentDTO.cs
namespace SafeDalat_API.Model.DTO.Icident
{
    public class CreateIncidentDTO
    {
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string Address { get; set; } = null!;
        public string Ward { get; set; } = null!;
        public string StreetName { get; set; } = null!;
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public int CategoryId { get; set; }

        public bool IsForceCreate { get; set; } = false;

        public double DeviceLatitude { get; set; }
        public double DeviceLongitude { get; set; }
    }
}
